install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```diff
@@ -785,11 +785,23 @@ spec:
       {{- end }}
         # To read the clustermesh configuration
       - name: clustermesh-secrets
-        secret:
-          secretName: cilium-clustermesh
+        projected:
           # note: the leading zero means this number is in octal representation: do not remove it
           defaultMode: 0400
-          optional: true
+          sources:
+          - secret:
```

Comment:
What exactly does this entry do since it has no `items`?

Reply:
All entries are automatically projected with the key as filename, if `items` is not specified. From the official docs:

> if unspecified, each key-value pair in the Data field of the referenced Secret will be projected into the volume as a file whose name is the key and content is the value. If specified, the listed keys will be projected into the specified paths, and unlisted keys will not be present.

Reply:
Should we specify them anyways? I assume we're expecting them in a particular format, this would ensure the secret mount errors if the secret is using the wrong keys.

Reply:
We cannot list them, since the keys of that secret change based on the number and name of the clusters to peer with. For each cluster, one key matches the cluster name and contains the etcd config, while three additional keys store the key, cert and ca-cert (if the CA is not shared). Specifying them there would cause the restart of all cilium agents if a peer gets added/removed.

Reply:
I see. Could you add a comment along those lines? I think it would help clarify the intent of the mount.

Reply:
For completeness, an agent restart is nonetheless necessary when using IPs as etcd endpoints, given that the host aliases need to be updated. That is not required, though, when targeting DNS names directly.